src/analysis/von_neumann_algebra/finite_dimensional.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import analysis.von_neumann_algebra.basic
+
+/-!
+
+# Finite-dimensional von Neumann algebras
+
+In this file, we prove some results in finite-dimensional von Neumann algebras.
+
+## Notation
+
+We let `V` be an inner product space over `ℂ`.
+
+## Main results
+
+* `idempotent_operator_in_vN_iff_ker_and_range_are_commutant_invariant`:
+idempotent `e ∈ L(V)` is in von Neumann algebra `M` if and only if
+`e.range,e.ker` are `M'` (i.e., commutant of `M` or `M.centralizer`)
+invariant subspaces
+-/
+
+variables {V : Type*} [inner_product_space ℂ V]
+
+-- this definition and lemma are copied from
+-- `analysis/inner_product_space/finite_dimensional (PR #18041)`
+/-- `U` is `T` invariant: `∀ u : V`, if `u ∈ U` then `T u ∈ U`-/
+def invariant_subspace (U : submodule ℂ V) (T : V →L[ℂ] V) : Prop := U ≤ U.comap T
+lemma invariant_subspace_def (U : submodule ℂ V) (T : V →L[ℂ] V) :
+invariant_subspace U T ↔ T '' U ⊆ U := by simp [set.image_subset_iff]; refl
+
+open continuous_linear_map
+
+-- Not sure if there already exists a version of this lemma in Mathlib.
+/-- given any idempotent operator `T ∈ L(V)` and element `x ∈ V`,
+there exists `v ∈ T.ker` and `w ∈ T.range` such that `x = v + w` -/
+lemma idempotent_operator_exists_ker_add_range
+(T : V →L[ℂ] V) (h : T^2 = T) (x : V) :
+∃ v : T.ker, ∃ w : T.range, x = v + w :=
```

Comment:
This might be better stated as
```suggestion
  codisjoint T.ker T.range :=
```

Reply:
How about if I add the following two lemmas instead?

one to prove that the kernel and range of any idempotent operator is codisjoint:
```
lemma idempotent_codisjoint_ker_range (T : V →L[ℂ] V) (h : T^2 = T) :
  codisjoint T.ker T.range :=
```
and one to prove the lemma I need from any two subspaces being codisjoint:
```
lemma codisjoint_exists_ker_add_range (T : V →L[ℂ] V) (U W : submodule ℂ V) (h : codisjoint U W) (x : V) : 
  ∃ u : U, ∃ w : W, x = u + w :=
```

Reply:
Is it the case that `is_compl T.ker T.range` which would be a stronger result?

Reply:
Yeah! Changed it now - thanks.